assets/js/modules/analytics-4/datastore/enhanced-measurement.js

```diff
@@ -427,6 +442,46 @@ const baseSelectors = {
 
 		return ! isEqual( settings, savedSettings );
 	},
+
+	/**
+	 * This is a utility selector that checks if either the GA4 module settings, or the
+	 * current web data stream's enhanced measurement settings have changed.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {boolean} True if settings have changed, otherwise false.
+	 */
+	haveAnyGA4SettingsChanged: createRegistrySelector( ( select ) => () => {
```

Comment:
It was necessary to introduce this new utility selector as this dual check needs to be made in a number of places...